keyboards/nuphy/air75_v2/ansi/ansi.c

```diff
@@ -0,0 +1,272 @@
+/*
+Copyright 2023 @ Nuphy <https://nuphy.com/>
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "user_kb.h"
+#include "ansi.h"
+#include "usb_main.h"
+
+extern bool f_rf_sw_press;
+extern bool f_sleep_show;
+extern bool f_dev_reset_press;
+extern bool f_bat_num_show;
+extern bool f_rgb_test_press;
+extern bool f_bat_hold;
+extern uint16_t  no_act_time;
+extern uint8_t   rf_sw_temp;
+extern uint16_t  rf_sw_press_delay;
+extern uint16_t  rf_linking_time;
+extern user_config_t user_config;
+extern DEV_INFO_STRUCT dev_info;
+
+
+/* qmk process record */
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
```

Comment:
I recently got a keyboard and forked nuphy repo while this PR is open in order to build my custom firmware. 

I noticed I was unable to customize some behavior because `ansi.c` is defining the `_user` versions of functions instead of the `_kb` ones [as recommended in the documentation](https://docs.qmk.fm/#/custom_quantum_functions?id=a-word-on-core-vs-keyboards-vs-keymap).

Reply:
@nuphy-src
It's related to this comment made by Drashna way before which still holds up.
https://github.com/qmk/qmk_firmware/pull/22751#discussion_r1436170338

Reply:
@nuphy-src just to clarify what i think @adophoxia is trying to convey:
QMK has a [hierarchy](https://github.com/qmk/qmk_firmware/blob/master/docs/custom_quantum_functions.md#a-word-on-core-vs-keyboards-vs-keymap-ida-word-on-core-vs-keyboards-vs-keymap) for some of their functions.

Since you've used the `_user` version of the `process_record_user` here I don't think it can be defined again in the keymap directory. Assuming they're suggesting switching this and other functions like it to `process_record_kb` and having a call to `process_record_user` first then doing this switch ([example](https://github.com/qmk/qmk_firmware/blob/3495cbc6382d4fa08ee5b4294527ab66fb7dff59/keyboards/keychron/q1v2/q1v2.c#L42-L46))